scikeras/wrappers.py

```diff
@@ -51,94 +41,67 @@ class BaseWrapper(BaseEstimator):
-    `batch_size` or `epochs` as well as the model parameters.
+            Used to build the Keras Model. When called,
+            must return a compiled instance of a Keras Model
+            to be used by `fit`, `predict`, etc.
+        random_state : int, RandomState instance, default=None
+            Set the Tensorflow random number generators to a
+            reproducible deterministic state using this seed.
+            Pass an int for reproducible results across multiple function calls.
+        For all other parameters see tf.keras.Model documentation.
     """
 
-    # basic legal parameter set, based on functions that will normally be
-    # called the model building function will be dynamically added
-    _legal_params_fns = [
-        Sequential.evaluate,
-        Sequential.fit,
-        Sequential.predict,
-        Model.evaluate,
-        Model.fit,
-        Model.predict,
-    ]
-
-    _sk_params = None
     is_fitted_ = False
 
     _tags = {
         "poor_score": True,
         "multioutput": True,
     }
 
-    def __init__(self, build_fn=None, **sk_params):
+    def __init__(
+        self,
+        build_fn=None,
+        *,
+        random_state=None,
+        optimizer="rmsprop",
+        loss=None,
+        metrics=None,
+        run_eagerly=None,
+        batch_size=None,
+        verbose=1,
+        steps=None,
+        callbacks=None,
+        epochs=1,
+        validation_split=0.0,
+        shuffle=True,
+        initial_epoch=0,
+        **kwargs,
+    ):
+        # Get defaults from `build_fn`
+        if inspect.isfunction(build_fn):
+            vars(self).update(_get_default_args(build_fn))
 
         if isinstance(build_fn, Model):
             # ensure prebuilt model can be serialized
             make_model_picklable(build_fn)
 
+        # Parse hardcoded params
         self.build_fn = build_fn
-
-        if inspect.isfunction(build_fn):
-            kwargs = _get_default_args(build_fn)
-            sk_params = {**kwargs, **sk_params}
-
-        if sk_params:
-            # for backwards compatibility
-
-            # the sklearn API requires that all __init__ parameters be saved
-            # as an instance attribute of the same name
-            for name, val in sk_params.items():
-                setattr(self, name, val)
-
-            # save keys so that we can count these as __init__ params
-            self._sk_params = list(sk_params.keys())
+        self.random_state = random_state
+        self.optimizer = optimizer
+        self.loss = loss
+        self.metrics = metrics
+        self.run_eagerly = run_eagerly
+        self.batch_size = batch_size
+        self.verbose = verbose
+        self.callbacks = callbacks
+        self.epochs = epochs
+        self.validation_split = validation_split
+        self.shuffle = shuffle
+        self.initial_epoch = initial_epoch
+
+        # Unpack kwargs
+        vars(self).update(**kwargs)
```

Comment:
Also borrowed from scikeras. So much cleaner with the new `_get_param_names` implementation.